code/modules/clothing/spacesuits/flightsuit.dm

```diff
@@ -0,0 +1,1100 @@
+	var/disabled = 0	//Whether it is disabled from crashes/emps/whatever
+	var/crash_disable_message = 0	//To not spam the user with messages
+	var/emp_disable_message = 0
+
+	//This is probably too much code just for EMP damage.
+	var/emp_damage = 0	//One hit should make it hard to control, continuous hits will cripple it and then simply shut it off/make it crash. Direct hits count more.
+	var/emp_strong_damage = 1.5
+	var/emp_weak_damage = 1
+	var/emp_heal_amount = 0.1		//How much emp damage to heal per process.
+	var/emp_disable_threshold = 3	//3 weak ion, 2 strong ion hits.
+	var/emp_disabled = 0
+
+	var/crash_damage = 0	//Same thing, but for crashes. This is in addition to possible amounts of brute damage to the wearer.
+	var/crash_damage_low = 1
+	var/crash_damage_high = 2.5
+	var/crash_disable_threshold = 5
+	var/crash_heal_amount = 0.2
+	var/crash_disabled = 0
+
+	var/datum/effect_system/trail_follow/ion/flight/ion_trail
+
+	var/assembled = 0
+	var/obj/item/weapon/stock_parts/manipulator/part_manip = null
+	var/obj/item/weapon/stock_parts/scanning_module/part_scan = null
+	var/obj/item/weapon/stock_parts/capacitor/part_cap = null
+	var/obj/item/weapon/stock_parts/micro_laser/part_laser = null
+	var/obj/item/weapon/stock_parts/matter_bin/part_bin = null
+
+
+//Start/Stop processing the item to use momentum and flight mechanics.
+/obj/item/device/flightpack/New()
+	START_PROCESSING(SSfastprocess, src)
+	..()
+	ion_trail = new
+	ion_trail.set_up(src)
+
+/obj/item/device/flightpack/full/New()
+	..()
+	part_manip = new /obj/item/weapon/stock_parts/manipulator/pico
+	part_scan = new /obj/item/weapon/stock_parts/scanning_module/phasic
+	part_cap = new /obj/item/weapon/stock_parts/capacitor/super
+	part_laser = new /obj/item/weapon/stock_parts/micro_laser/ultra
+	part_bin = new /obj/item/weapon/stock_parts/matter_bin/super
+	assembled = 1
+
+/obj/item/device/flightpack/proc/update_parts()
+	boost_chargerate = initial(boost_chargerate)
+	boost_drain = initial(boost_drain)
+	powersetting_high = initial(powersetting_high)
+	emp_disable_threshold = initial(emp_disable_threshold)
+	crash_disable_threshold = initial(crash_disable_threshold)
+	stabilizer_decay_amount = initial(stabilizer_decay_amount)
+	airbrake_decay_amount = initial(airbrake_decay_amount)
+	var/manip = 0	//Efficiency
+	var/scan = 0	//Damage avoidance/other
+	var/cap = 0		//Charging
+	var/laser = 0	//Power
+	var/bin = 0		//Stability
+	assembled = 0	//Ready?
+	if(part_manip && part_scan && part_cap && part_laser && part_bin)
+		manip = part_manip.rating
+		scan = part_scan.rating
+		cap = part_cap.rating
+		laser = part_laser.rating
+		bin = part_bin.rating
+		assembled = 1
+	boost_chargerate *= cap
+	boost_drain -= manip
+	powersetting_high = Clamp(laser, 0, 3)
+	emp_disable_threshold = bin*1.25
+	crash_disable_threshold = bin*2
+	stabilizer_decay_amount = scan*5.75
+	airbrake_decay_amount = manip*15
+
+/obj/item/device/flightpack/Destroy()
+	if(suit)
+		delink_suit()
+	qdel(part_manip)
```

Comment:
null these references after you call qdel on them

    part_manip = null
    ...